### PR TITLE
gh-107619: Extend functools LRU cache docs with generators and async functions

### PR DESCRIPTION
**Pull request by Hels15**
Extend `functools LRU cache` docs with `generators and async functions`.  An async function `returns a coroutine, not the actual function result`. That means that the cache decorator is caching the coroutine, not the function result. Since the coroutine is not "reawaitable", we get an exception. This pr `includes generators and async functions in the list of places where it doesn't cache doesn't make sense`.

* Issue: gh-107619

----
:books: Documentation preview :books:: https://cpython-previews--107934.org.readthedocs.build/

### CODE REVIEW

**Review by rhettinger — APPROVED** [review]

File: Doc/library/functools.rst [hunk 1]
```diff
@@ -226,7 +226,7 @@ The :mod:`functools` module defines the following functions:
 
    In general, the LRU cache should only be used when you want to reuse
    previously computed values.  Accordingly, it doesn't make sense to cache
-   functions with side-effects, functions that need to create distinct mutable
+   functions with side-effects, generators and async functions that need to create distinct mutable
```
Comment (AA-Turner) [inline Doc/library/functools.rst:229]: This wraps the lines:
```suggestion
   functions with side-effects, generators and async functions that
   need to create distinct mutable objects on each call,
   or impure functions such as time() or random().
```

File: Doc/library/functools.rst [hunk 2]
```diff
@@ -226,7 +226,7 @@ The :mod:`functools` module defines the following functions:
...
    objects on each call, or impure functions such as time() or random().
```
Comment (AA-Turner) [inline Doc/library/functools.rst]: This wraps the lines:
```suggestion
   functions with side-effects, generators and async functions that
   need to create distinct mutable objects on each call,
   or impure functions such as time() or random().
```

**Comment by Hels15** [comment]
Accepted the suggested changes.

File: Doc/library/functools.rst [hunk 3]
```diff
@@ -226,8 +226,9 @@ The :mod:`functools` module defines the following functions:
 
    In general, the LRU cache should only be used when you want to reuse
    previously computed values.  Accordingly, it doesn't make sense to cache
-   functions with side-effects, functions that need to create distinct mutable
-   objects on each call, or impure functions such as time() or random().
+   functions with side-effects, generators and async functions that
+   need to create distinct mutable objects on each call,
```
Comment (AA-Turner) [inline Doc/library/functools.rst]: I think there's a change of meaning here, by inserting the words "generators and async" you've curtailed the possibility of standard functions creating mutable objects.

Perhaps:

```suggestion
   functions with side-effects, functions that need to create
   distinct mutable objects on each call (such as generators and async functions),
```

**Comment by Hels15** [comment]
Oh yes, you are right, I have added what you said.

**Review by AA-Turner — APPROVED** [review]

File: Doc/library/asyncio-eventloop.rst [hunk 4]
```diff
@@ -1442,6 +1442,7 @@ async/await code consider using the high-level
    * *stdin* can be any of these:
 
      * a file-like object
+     * an existing file descriptor (a positive integer), for example those created with :meth:`os.pipe()`
```
Comment (kumaraditya303) [inline Doc/library/asyncio-eventloop.rst]:
```suggestion
```

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**